src/wifi.cpp

```diff
@@ -0,0 +1,694 @@
+// is no longer readable, it automatically kills itself and resets the handle.
+void task_read_ip_packets()
+{
+    // Create a 0-initialized buffer IP packets will be read into
+    char read_buffer[16] = { 0 };
+    int num_read_bytes = 0;
+
+    while(1)
+    {
+        // Get the number of bytes in the file descriptor,
+        // copy up to count bytes from the file descriptor to our read buffer
+        // https://man7.org/linux/man-pages/man2/read.2.html
+        num_read_bytes = read(
+            /* int fd = */ ip_socket_file_descriptor,
+            /* void buf[.count] = */ read_buffer,
+            /* size_t count = */ sizeof(read_buffer) - sizeof('\n'));
+        if(0 > num_read_bytes)
+        {
+            // Failed to read the file descriptor, stop this task and free its resources
+            s_println("Failed to read IP packet file descriptor, stopping task to read IP packets");
+            (void) close(/* int fd = */ ip_socket_file_descriptor);
+            ip_socket_file_descriptor = 0;
+            read_ip_packet_task_handle = nullptr;
+            vTaskDelete(/* TaskHandle_t xTaskToDelete = */ NULL);
+        }
+
+        // Print the bytes from our user input to stdout (replace ending \n with \0)
+        read_buffer[num_read_bytes - sizeof('\n')] = '\0';
+        s_println(read_buffer);
+
+        // See if the the packet matches a command, if so, execute it
+        for(size_t i = 0; i < NUM_TCP_COMMANDS; ++i)
+        {
+            if(((num_read_bytes - sizeof('\0')) == strlen(tcp_commands[i].command)) && 
+                (0 == strncmp(read_buffer, tcp_commands[i].command, num_read_bytes)))
+            {
+                (*(tcp_commands[i].action))();
+            }
+        }
+
+        // 17SEP2024: usStackDepth = 1024 + 512, uxTaskGetHighWaterMark = 420
+        PRINT_STACK_USAGE();
+    }
+}
+
+// ============================== //
+// Define code to connect to WiFi //
+// ============================== //
+
+static inline bool wifi_init();
+// bool wifi_free();
+static void event_any_wifi(
+    void *arg,
+    esp_event_base_t event_base,
+    int32_t event_id,
+    void *event_data);
+static void event_got_ip(
+    void *arg,
+    esp_event_base_t event_base,
+    int32_t event_id,
+    void *event_data);
+static inline bool wifi_event_group_init(
+    esp_event_handler_instance_t &event_handler_instance_wifi,
+    esp_event_handler_instance_t &event_handler_instance_ip);
+static bool wifi_event_group_free(
+    esp_event_handler_instance_t &event_handler_instance_wifi,
+    esp_event_handler_instance_t &event_handler_instance_ip);
+static inline bool wifi_connect(
+    char *wifi_ssid,
+    char *wifi_password);
+
+// If all went well, by the end of this function, the device will have IP address
+// and can connect to a socket to talk with other servers on a network
+bool wifi_start(
+    char *wifi_ssid,
+    char *wifi_password)
+{
+    // Allocate TCP/IP/WiFi resrouces
```

Comment:
resrouces -> resources

Reply:
addressed in: https://github.com/jCallon/squirt/pull/1/commits/edb88a636d2cf54a42f970d190cd054c04993412